### PR TITLE
fix Dense's docstring

### LINKED ISSUE #1422: Layers' docstrings and constructors inconsistencies
We have a pretty bad situation with the docstrings of Dense and Conv layers (and maybe others) not matching the corresponding method definitions.  

In Conv's docstrings we have
```julia
Conv(filter, in => out, σ=identity; stride=1, pad=0, dilation=1)
Conv(weight::AbstractArray, bias, [activation; stride, pad, dilation])
Conv(; weight, bias, [activation, stride, pad, dilation])
```
not perfectly aligned with the actual definitions:
```julia
Conv(k::NTuple{N,Integer}, ch::Pair{<:Integer,<:Integer}, σ = identity;
            init = glorot_uniform,  stride = 1, pad = 0, dilation = 1,
            weight = convfilter(k, ch, init = init), bias = true)
Conv(w::AbstractArray{T,N}, b::Union{Bool, Zeros, AbstractVector{T}}, σ = identity;
              stride = 1, pad = 0, dilation = 1) where {T,N}
Conv(; weight::AbstractArray{T,N}, bias::Union{Bool, Zeros, AbstractVector{T}},
              activation = identity, stride = 1, pad = 0, dilation = 1) where {T,N}
```

The situation is even worse for Dense. The docstring contains only
```julia
Dense(in::Integer, out::Integer, σ = identity; bias=true)
```
while the actual constructors are
```julia
Dense(W, b)
Dense(in::Integer, out::Integer, σ = identity; initW=glorot_uniform, initb=zeros, bias=true)
```

We should also deprecate `init` in Conv and introduce `initW` and `initb` as in the Dense layer.

### PR DESCRIPTION
**Pull request by CarloLucibello**
partial fix for #1422

### CODE REVIEW

File: src/layers/basic.jl [hunk 1]
```diff
@@ -99,18 +100,19 @@ dimension is `out` and the remaining dimensions are the same as in the input.
 
 Setting `bias` to `false` will switch the bias  off for the layer.
 
-# Example
-```
+`initW` and `initb` are callables used to initialize weights and biases respectively,
+trhough the calls `initW(out, in)` and `initb(out)`.
```
Comment (CarloLucibello) [inline src/layers/basic.jl:104]:
```suggestion
through the calls `initW(out, in)` and `initb(out)`.
```

File: src/layers/basic.jl [hunk 2]
```diff
@@ -83,7 +83,8 @@ extraChain(::Tuple{}, x) = ()
 
 
 """
-    Dense(in::Integer, out::Integer, σ = identity; bias=true)
+    Dense(in, out, σ=identity; initW=glorot_uniform, initb=zeros, bias=true)
```
Comment (gxyd) [inline src/layers/basic.jl:86]: Any specific reason to explicitly remove `in::Integer` and `out::Integer`?

Comment (CarloLucibello) [reply src/layers/basic.jl:86]: just to avoid having an extra-long line, I hope that the expected type is clear from the docstring

**Comment by CarloLucibello** [comment]
why bors didn't merge? maybe an approval is needed?

**Review by DhairyaLGandhi — APPROVED** [review]

**Comment by DhairyaLGandhi** [comment]
Let's try again

bors r+

**Pull request merged**